### ISSUE TITLE
Intel regression tester are failing for 3 cases

### ISSUE THREAD

**Issue by mkrack** [cp2k/cp2k#733]
The tests
```
- H2O_ec.inp
- HF_ec_dipole.inp
- HF_ec_field.inp
```
 fail with the Intel compiler version 18 as well as 19. The issue seems to be related to DBCSR and is persisting since a while. It is not clear if this is a compiler bug or a cp2k/dbcsr problem.

**Comment by alazzaro** [comment]
Same for the CCE test...
Actually, it is not strictly DBCSR related, rather cp_dbcsr...
It errors happens at this line:

https://github.com/cp2k/cp2k/blob/8a41e8745e366876208ac49623ec0b903d162ec6/src/cp_dbcsr_operations.F#L1438

with message

A pointer passed to DEALLOCATE points to an object that cannot be deallocated

Concerning the other failing tests on mini-arch, the tests are not protected when libint is not available...

**Comment by mkrack** [comment]
One case (HF_ec_field.inp), however, shows a deeper trace into DBCSR:
```
cp2k.psmp:67053 terminated with signal 11 at PC=3307639 SP=7fffb7ee2600.  Backtrace:

cp2k.psmp:67052 terminated with signal 11 at PC=3307639 SP=7fff027d7e80.  Backtrace:
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_dist_operations_mp_dbcsr_get_block_index_+0x29)[0x3307639]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_dist_operations_mp_dbcsr_get_block_index_+0x29)[0x3307639]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_block_access_mp_dbcsr_get_2d_block_p_d_+0x61)[0x3427501]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_block_access_mp_dbcsr_get_2d_block_p_d_+0x61)[0x3427501]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_api_mp_dbcsr_get_2d_block_notrans_p_d_+0x1c)[0x33137fc]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(dbcsr_api_mp_dbcsr_get_2d_block_notrans_p_d_+0x1c)[0x33137fc]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(ec_efield_local_mp_ec_efield_mo_derivatives_+0x1f7a)[0x1f396da]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(ec_efield_local_mp_ec_efield_mo_derivatives_+0x1f7a)[0x1f396da]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(ec_efield_local_mp_ec_efield_local_operator_+0x558)[0x1f37738]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(ec_efield_local_mp_ec_efield_local_operator_+0x558)[0x1f37738]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(energy_corrections_mp_ec_build_core_hamiltonian_force_+0x25be)[0x1b5545e]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(energy_corrections_mp_ec_build_core_hamiltonian_force_+0x25be)[0x1b5545e]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(energy_corrections_mp_energy_correction_+0x210f)[0x1b509ff]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(energy_corrections_mp_energy_correction_+0x210f)[0x1b509ff]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(qs_force_mp_qs_forces_+0x459f)[0x1824cef]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(qs_force_mp_qs_forces_+0x459f)[0x1824cef]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(qs_force_mp_qs_calc_energy_force_+0x227)[0x1820257]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(qs_force_mp_qs_calc_energy_force_+0x227)[0x1820257]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(force_env_methods_mp_force_env_calc_energy_force_+0x3e5)[0x153b4e5]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(force_env_methods_mp_force_env_calc_energy_force_+0x3e5)[0x153b4e5]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(cp2k_runs_mp_run_input_+0x14b7)[0x842097]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(cp2k_runs_mp_run_input_+0x14b7)[0x842097]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(MAIN__+0x861)[0x839151]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(MAIN__+0x861)[0x839151]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(main+0x2e)[0x8388de]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2ab06f884495]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp(main+0x2e)[0x8388de]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2abf3703c495]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp[0x8387af]
/data/user/krack/rt/intel-psmp/cp2k/exe/Linux-x86-64-intel-regtest/cp2k.psmp[0x8387af]
EXIT CODE:  1  MEANING:  RUNTIME FAIL
```

**Comment by alazzaro** [comment]
The problem it seems started after this commit: https://github.com/cp2k/cp2k/commit/63f2e15396cd4d5323c594035dca6ba31437099f

My guess is that we are looking for blocks that are not available in the particular processor, so the pointer is somehow null. I can think something wrong with the distribution...
In DBCSR it is possible to add a check for that, but still, the problem is elsewhere...

**Comment by alazzaro** [comment]
Is this issue still relevant?

**Comment by mkrack** [comment]
I think that was fixed indeed by #761.

**Comment by mkrack** [comment]
Fixed, see #761.
